data/types/base/pump/basal_schedule.go

```diff
@@ -33,14 +33,27 @@ func ParseBasalSchedule(parser data.ObjectParser) *BasalSchedule {
 	return basalSchedule
 }
 
-func ParseBasalScheduleArray(parser data.ArrayParser) *map[string][]*BasalSchedule {
-	var basalScheduleArray *map[string][]*BasalSchedule
+func ParseBasalScheduleArray(parser data.ArrayParser) *[]*BasalSchedule {
+	var basalScheduleArray *[]*BasalSchedule
 	if parser.Array() != nil {
```

Comment:
Since this function can return nil, do we validate that it isn't nil during the Validate phase?